lib/Parse/ParseStmt.cpp

```diff
@@ -880,9 +880,10 @@ ParserResult<Stmt> Parser::parseStmtYield(SourceLoc tryLoc) {
     }
 
     auto expr = parseExpr(diag::expected_expr_yield);
-    if (expr.hasCodeCompletion())
-      return makeParserCodeCompletionResult<Stmt>();
-    if (expr.isParseErrorOrHasCompletion()) {
+    if (expr.hasCodeCompletion()) {
+      status.setHasCodeCompletionAndIsError();
+      yields.push_back(expr.get());
```

Comment:
Could `expr` possibly be null here? Seems like some functions in the parser produce `makeParserCodeCompletionResult<Expr>()`.

Reply:
Good call. I added a check for `expr.isNonNull`